src/adapter/src/catalog/open.rs

```diff
@@ -1115,37 +1123,44 @@ fn default_logging_config() -> ReplicaLogging {
-    pub probe_cluster: String,
-    /// Size to default support_cluster on bootstrap
-    pub support_cluster: String,
+
+#[derive(Debug)]
+pub struct BuiltinBootstrapClusterConfigMap {
+    /// Size and replication factor to default system_cluster on bootstrap
+    pub system_cluster: BootstrapBuiltinClusterConfig,
+    /// Size and replication factor to default catalog_server_cluster on bootstrap
+    pub catalog_server_cluster: BootstrapBuiltinClusterConfig,
+    /// Size and replication factor to default probe_cluster on bootstrap
+    pub probe_cluster: BootstrapBuiltinClusterConfig,
+    /// Size and replication factor to default support_cluster on bootstrap
+    pub support_cluster: BootstrapBuiltinClusterConfig,
     /// Size to default analytics_cluster on bootstrap
-    pub analytics_cluster: String,
+    pub analytics_cluster: BootstrapBuiltinClusterConfig,
 }
 
-impl BuiltinBootstrapClusterSizes {
+impl BuiltinBootstrapClusterConfigMap {
     /// Gets the size of the builtin cluster based on the provided name
-    fn get_size(&self, cluster_name: &str) -> Result<String, mz_catalog::durable::CatalogError> {
-        if cluster_name == mz_catalog::builtin::MZ_SYSTEM_CLUSTER.name {
-            Ok(self.system_cluster.clone())
-        } else if cluster_name == mz_catalog::builtin::MZ_CATALOG_SERVER_CLUSTER.name {
-            Ok(self.catalog_server_cluster.clone())
-        } else if cluster_name == mz_catalog::builtin::MZ_PROBE_CLUSTER.name {
-            Ok(self.probe_cluster.clone())
-        } else if cluster_name == mz_catalog::builtin::MZ_SUPPORT_CLUSTER.name {
-            Ok(self.support_cluster.clone())
-        } else if cluster_name == mz_catalog::builtin::MZ_ANALYTICS_CLUSTER.name {
-            Ok(self.analytics_cluster.clone())
-        } else {
-            Err(mz_catalog::durable::CatalogError::Catalog(
-                SqlCatalogError::UnexpectedBuiltinCluster(cluster_name.to_owned()),
-            ))
-        }
+    fn get_config(
+        &self,
+        cluster_name: &str,
+    ) -> Result<BootstrapBuiltinClusterConfig, mz_catalog::durable::CatalogError> {
+        let cluster_config = match cluster_name {
+            name if name == mz_catalog::builtin::MZ_SYSTEM_CLUSTER.name => &self.system_cluster,
+            name if name == mz_catalog::builtin::MZ_CATALOG_SERVER_CLUSTER.name => {
+                &self.catalog_server_cluster
+            }
+            name if name == mz_catalog::builtin::MZ_PROBE_CLUSTER.name => &self.probe_cluster,
+            name if name == mz_catalog::builtin::MZ_SUPPORT_CLUSTER.name => &self.support_cluster,
+            name if name == mz_catalog::builtin::MZ_ANALYTICS_CLUSTER.name => {
+                &self.analytics_cluster
+            }
+            _ => {
+                return Err(mz_catalog::durable::CatalogError::Catalog(
+                    SqlCatalogError::UnexpectedBuiltinCluster(cluster_name.to_owned()),
+                ))
+            }
+        };
```

Comment:
nit: it feels like this would be better written as if-else statements? If you want to prevent all of the `Ok` wrapping you can do:
```
let cluster_config = if cluster_name == "foo" {
    &self.foo_cluster
} else {
    return ...
}
```

Reply:
sgtm!